### PR TITLE
OPENEUROPA-1242: Assign user roles in Drupal on load

### PR DESCRIPTION
**Pull request by upchuk**
### Description

Assign user roles in Drupal on load

### CODE REVIEW

File: modules/oe_authorisation_syncope/oe_authorisation_syncope.services.yml [hunk 1]
```diff
@@ -7,4 +7,4 @@ services:
     arguments: ['@oe_authorisation_syncope.client', '@entity_type.manager', '@logger.factory', '@state']
   oe_authorisation_syncope.user_mapper:
     class: Drupal\oe_authorisation_syncope\SyncopeUserMapper
-    arguments: ['@oe_authorisation_syncope.client', '@oe_authorisation_syncope.role_mapper', '@logger.factory']
\ No newline at end of file
+    arguments: ['@oe_authorisation_syncope.client', '@oe_authorisation_syncope.role_mapper', '@logger.factory', '@entity_type.manager']
```
Comment (ademarco) [inline modules/oe_authorisation_syncope/oe_authorisation_syncope.services.yml]: Missing new line.

File: modules/oe_authorisation_syncope/tests/src/Kernel/SyncopeUserTest.php [hunk 2]
```diff
@@ -18,25 +18,25 @@ class SyncopeUserTest extends SyncopeTestBase {
    * Ensures the corresponding actions take place in Syncope.
    */
   public function testUserLifeCycle(): void {
-    $user = $this->createUser('jack');
+    $user = $this->createUser('Kevin');
```
Comment (drupol) [inline modules/oe_authorisation_syncope/tests/src/Kernel/SyncopeUserTest.php]: All of the changes in this file are unrelated and out of scope.

File: tests/features/syncope.feature [hunk 3]
```diff
@@ -20,15 +20,15 @@ Feature: Syncope integration
...
+  Scenario Outline: Users should lose their roles in Drupal on login if they no longer have them assigned in Syncope
     Given users:
       | name | mail             |
-      | jack | jack@example.com |
-    And the user "jack" has the roles "Site Manager, Support Engineer" in Drupal
-    And the user "jack" loses the role "<lost>" in Syncope
-    And I am logged in as "jack"
-    Then the user "jack" should not have the role "<lost>" in Drupal
-    And the user "jack" should have the role "<kept>" in Drupal
+      | Kevin | Kevin@example.com |
```
Comment (drupol) [inline tests/features/syncope.feature]: The changes related to the first name change from Jack to Kevin are unrelated and out of scope.

**Comment by voidtek** [comment]
The grumphp tests are red

```
+ ./vendor/bin/grumphp run
GrumPHP is sniffing your code!
Running task 1/2: Phpcs... ✘
Running task 2/2: PhpMd... ✔
FILE: ...modules/oe_authorisation_syncope/oe_authorisation_syncope.module
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 14 | WARNING | [x] Unused use statement
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 2.44 secs; Memory: 12Mb

You can fix some errors by running following command:
'/test/oe_authorisation/vendor/bin/phpcbf' '--standard=./vendor/drupal/coder/coder_sniffer/Drupal' '--report=full' '--ignore=vendor/,node_modules/,build/' '/test/oe_authorisation/modules/oe_authorisation_syncope/oe_authorisation_syncope.module'
To skip commit checks, add -n or --no-verify flag to commit command
```

File: modules/oe_authorisation_syncope/src/SyncopeClient.php [hunk 4]
```diff
@@ -361,7 +361,10 @@ public function createUser(SyncopeUser $user, $destination_realm = 'site'): Sync
       $groups[] = $membership->groupKey;
     }
 
-    return new SyncopeUser($object->key, $object->name, $groups);
+    $syncope_user = new SyncopeUser($object->key, $object->name, $groups);
+    $date = new \DateTime($object->lastChangeDate);
+    $syncope_user->setUpdated($date->getTimestamp());
```
Comment (voidtek) [inline modules/oe_authorisation_syncope/src/SyncopeClient.php]: Why we are not filling the update date during the _construct?

We are writing 4 times this code.
```
$syncope_user = new SyncopeUser($object->key, $object->name, $groups);
    $date = new \DateTime($object->lastChangeDate);
    $syncope_user->setUpdated($date->getTimestamp());
    return $syncope_user;
```

Comment (upchuk) [reply modules/oe_authorisation_syncope/src/SyncopeClient.php]: @voidtek Because we only need this when it comes back from SYncope. And the only line we'd be saving is:

```
$syncope_user->setUpdated($date->getTimestamp());
```

**Comment by upchuk** [comment]
@ademarco @voidtek I fixed the grump thing. Was surprised about it because locally it was fine but then I saw Adam has pushed something :)
